pkgs/development/python-modules/luma-core/default.nix

```
@@ -0,0 +1,52 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  setuptools,
+  pillow,
+  smbus2,
+  pyftdi,
+  cbor2,
+  rpi-gpio,
+  spidev,
+}:
+buildPythonPackage rec {
+  pname = "luma-core";
+  version = "2.5.1";
+
+  src = fetchPypi {
+    pname = "luma_core";
+    inherit version;
+    hash = "sha256-m4j76GiYnsRHfiEP4pk+d0p0Jdfeg3iBQQn5i+MsplM=";
+  };
+
+  build-system = [ setuptools ];
+  pyproject = true;
+
+  dependencies = [
+    pillow
+    smbus2
+    cbor2
+  ] ++ optional-dependencies.complete;
+
+  optional-dependencies = lib.fix (self: {
+    gpio = [ rpi-gpio ];
+    spi = [ spidev ];
+    ftdi = [ pyftdi ];
+    complete = self.gpio ++ self.spi ++ self.ftdi;
+  });
```

no need to use lib.fix, as the top-level attrs are recursive:
```suggestion
  optional-dependencies = {
    gpio = [ rpi-gpio ];
    spi = [ spidev ];
    ftdi = [ pyftdi ];
    complete = with optional-dependencies; gpio ++ spi ++ ftdi;
  };
```